gpii/node_modules/accessRequester/src/accessRequester.js

```diff
@@ -0,0 +1,95 @@
+/*!
+GPII Access Requester
+
+Copyright 2017 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = fluid || require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    $ = fluid.registerNamespace("jQuery");
+
+/*
+ * Access Requester provides getAccessToken(userToken) API that returns a promise object whose resolved value is the access token.
+ * When this API is called, the access requester does:
+ * 1. Retrieves client credentials;
+ * 2. Communicate with GPII Cloud end point via [resource owner GPII token grant](https://wiki.gpii.net/w/GPII_OAuth_2_Guide#Resource_Owner_GPII_Token_Grant);
+ * 3. Retrieves and returns a promise object whose resolved value contains the access token responded by the GPII Cloud /access_token endpoint;
+ *
+ * Access Requester requires these input options:
+ * @url {String} The GPII cloud http end point to request access tokens granted by OAuth2 resource owner GPII token grant.
+ * such as https://flowmanager.gpii.net/access_token
+ * @clientCredentialDataSourceGrade {String or Array of Strings} The grade name of the implementation of "clientCredentialDataSource" subcomponent
+ * that provides a get() API that returns a promise object whose resolved value is the client credential.
+ */
+fluid.defaults("gpii.accessRequester", {
+    gradeNames: ["fluid.component"],
+
+    // These options must be provided by integrators
+    url: "/access_token",    // Must be provided by integrators. The API endpoint provided by the authorization server to request access tokens via Resource Owner GPII Token Grant
+    clientCredentialDataSourceGrade: null,    // Must be provided by integrators
+    // End of integrators provided options
+
+    distributeOptions: {
+        source: "{that}.options.clientCredentialDataSourceGrade",
+        target: "{that > clientCredentialDataSource}.options.gradeNames"
+    },
+    components: {
+        // Must be provided by integrators
+        clientCredentialDataSource: {
+            type: "fluid.component"
+        },
+        accessTokenDataSource: {
+            type: "kettle.dataSource.URL",
+            options: {
+                url: "{accessRequester}.options.url",
+                writable: true,
+                writeMethod: "POST",
+                dataSourceModel: {
+                    grant_type: "password",
+                    username: "{accessRequester}.options.userToken",
```

Comment:
How is `"{accessRequester}.options.userToken"` set?

Reply:
Good catch. This was from the first draft of the implementation. `userToken` was later decided to be passed in as a function parameter. This line ended up not in use and is always replaced by [the parameter construction code](https://github.com/cindyli/universal/blob/3ec15c8e347765d8745dbc32a5199b1822e9c35b/gpii/node_modules/accessRequester/src/accessRequester.js#L77) in the `getAccessToken` function.